Library/Homebrew/tab.rb

```diff
@@ -70,35 +88,58 @@ def self.create(formula, compiler, stdlib)
           "version_scheme" => formula.version_scheme,
         },
       },
-      "built_on"                => DevelopmentTools.build_system_info,
-    }
+    })
 
     # We can only get `tap_git_head` if the tap is installed locally
     attributes["source"]["tap_git_head"] = formula.tap.git_head if formula.tap&.installed?
 
-    new(attributes)
+    new(attributes, type: :formula)
+  end
+
+  def self.create_from_cask(cask)
+    attributes = generic_attributes.merge({
+      "tabfile"              => cask.metadata_main_container_path/FILENAME,
+      "loaded_from_api"      => cask.loaded_from_api?,
+      "caskfile_only"        => cask.caskfile_only?,
+      "runtime_dependencies" => Tab.cask_runtime_deps_hash(cask, cask.depends_on),
+      "source"               => {
+        "path"         => cask.sourcefile_path.to_s,
+        "tap"          => cask.tap&.name,
+        "tap_git_head" => nil, # Filled in later if possible
+        "version"      => cask.version.to_s,
+      },
+      "uninstall_artifacts"  => cask.artifacts_list(uninstall_phase_only: true),
+    })
+
+    # We can only get `tap_git_head` if the tap is installed locally
+    attributes["source"]["tap_git_head"] = cask.tap.git_head if cask.tap&.installed?
+
+    new(attributes, type: :cask)
   end
 
   # Returns the {Tab} for an install receipt at `path`.
   #
   # NOTE: Results are cached.
-  def self.from_file(path)
+  def self.from_file(path, type: :formula)
     cache.fetch(path) do |p|
       content = File.read(p)
-      return empty if content.blank?
+      return empty(type:) if content.blank?
 
-      cache[p] = from_file_content(content, p)
+      cache[p] = from_file_content(content, p, type:)
     end
   end
 
   # Like {from_file}, but bypass the cache.
-  def self.from_file_content(content, path)
+  def self.from_file_content(content, path, type: :formula)
```

Comment:
Think similarly this would be better being two methods: `from_formula_file_content` or `from_cask_file_content` and we `odeprecate` `from_file_content`